pkg/machines/libvirt-dbus.js

```diff
@@ -800,11 +804,55 @@ const LIBVIRT_DBUS_PROVIDER = {
                             dispatch(updateVm(Object.assign({}, props, dumpxmlParams)));
                         else
                             dispatch(updateOrAddVm(Object.assign({}, props, dumpxmlParams)));
+
+                        dispatch(getVmSnapshots({ connectionName, domainPath: objPath }));
                     })
                     .catch(ex => console.warn("GET_VM action failed for path", objPath, ex.toString()));
         };
     },
 
+    GET_DOMAIN_SNAPSHOTS({ connectionName, domainPath }) {
+        return dispatch => call(connectionName, domainPath, 'org.libvirt.Domain', 'ListDomainSnapshots', [0], { timeout, type: 'u' })
+                .then(objPaths => {
+                    const snaps = [];
+                    const promises = [];
+
+                    objPaths[0].forEach(objPath => promises.push(call(connectionName, objPath, 'org.libvirt.DomainSnapshot', 'GetXMLDesc', [0], { timeout, type: 'u' })));
+
+                    // WA to avoid Promise.all() fail-fast behavior
+                    const toResultObject = (promise) => {
+                        return promise
+                                .then(result => ({ success: true, result }))
+                                .catch(error => ({ success: false, error }));
+                    };
+
+                    Promise.all(promises.map(toResultObject))
+                            .catch(ex => console.warn("DomainSnapshot method GetXMLDesc failed", ex.toString()))
```

Comment:
Can this happen in practice? toResultObject() maps all of them to a successful return, doesn't it?

Reply:
As mentioned, now I'm not sure now. I think it will not happen too, but that's just my guess